app/auth/views/users.py

```diff
@@ -341,3 +341,24 @@ def get_user_profile():
         user['supplier'] = supplier
 
     return jsonify(user=user), 200
+
+
+@auth.route('/profile', methods=['PATCH'])
+@login_required
+def update_user_profile():
+    json_payload = get_json_from_request()
+    role = None if current_user.is_anonymous else current_user.role
+
+    if role is not 'supplier':
```

Comment:
if they have no supplier, we just wouldn't update it and return back the profile with 200